app/services/defect_filter.rb

```diff
@@ -34,6 +36,20 @@ def scheme_scope
     [:for_scheme, schemes]
   end
 
+  def escalation_scope
```

Comment:
I'm wondering if these might be more expressive with a `case` statement, rather than having all these `returns`, or just a simple `if..elsif..else` statement, like this:

```ruby
if manually_escalated?
  escalated_permutations
elsif due_soon? && overdue?
  :overdue_and_due_soon
elsif overdue?
  :overdue
elsif due_soon?
  :due_soon
end
```

Reply:
I went back and forth on a few different ways to do this, I'm glad this was flagged in review because I wasn't particularly happy with all the returns either. Thanks!